### PR TITLE
CRR - Friendship v1 - Social Media APP

### PR DESCRIPTION
**Pull request by kevcap**
## This project was implemented with the following features:
- [x] Associations (has_many, belongs_to)
- [x] Friendship request (send, receive, accept, reject)
- [x] Invite users at user list or single page user.
- [x] Tests with capybara and rspec
- [x] CSS validation with rubocop
### Plus: 
✨Like counter
✨Logged username

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -6,4 +6,33 @@
...
+          <% if current_user.friend?(user) %>
+            <button><%= link_to 'Unfriend', remove_friend_path(user_id: user.id) , method: :delete , data: {confirm: "Are you sure you want to remove #{user.name} from your friend list?"}  %></button> 
+          <% elsif @pending_requests.include?(user) %>
+            <button><%= link_to 'Request Pending', '#' %></button> 
+
+          <% elsif user != current_user %>
+            <button><%= link_to 'Request friendship', invitations_path(friend_id: user.id) , method: :post %></button> 
+          <% end %>
+
+        <% end %>
```
Comment (maelfosso) [inline app/views/users/show.html.erb:33]: Please, logic are not allowed in views. Implement it into views helpers

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+          <% if current_user.friend?(user) %>
+            <button><%= link_to 'Unfriend', remove_friend_path(user_id: user.id) , method: :delete , data: {confirm: "Are you sure you want to remove #{user.name} from your friend list?"}  %></button> 
+          <% elsif @pending_requests.include?(user) %>
+            <button><%= link_to 'Request Pending', '#' %></button> 
+
+          <% elsif user != current_user %>
+            <button><%= link_to 'Request friendship', invitations_path(friend_id: user.id) , method: :post %></button> 
+          <% end %>
+
+        <% end %>
```
Comment (maelfosso) [inline app/views/users/index.html.erb:23]: Remove the logic from views, please. Implement them in views helpers

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for the well done job so far
However, there are some issues to have to be solved

1. It's asked, read it [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png), the posts timeline must include posts written by logged-in user and their friends. It's not implemented like this in your webapp

2. It's required that, read it it[here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png), there should be a button to invite user to friendship on users list page and on single user page. In your application it only appears on the Users list page not on Single page. Add it, please.

3. REad the comments below.

4. Add integration tests for **friendship request**

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

**Comment by kevcap** [comment]
# Hey @maelfosso 👋 I tried to talk with you through slack to check that points below, but I didn't get any answer 😕.

## Summing up I'm afraid that there are some issues on that review, please check down below:
"It's required that, read it there, there should be a button to invite user to friendship on users list page and on single user page. In your application, it only appears on the Users list page not on Single page. Add it, please."

### However I already have that feature as you can see below.
image
### About the friendship test, I added it as a unit test just as the requirement says.  It doesn't have to be an integrated spec test. I also have integrated tests, btw.
image

Regards!

**Review by rOluochKe — APPROVED** [review]
Hello @kevincappuccino 
### You did well :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**